CodeHood/bgtracker
# Ch monorepo setup

### What does this PR do?
This PR adds initial monorepo settings with a client-side app initialized in Next and a server app initialized in Nest.

### Description of Task completed
closes #3 

### How should this be manually tested?
- clone the repo `git clone git@github.com:CodeHood/bgtracker.git`
- git checkout to `ch-monorepo-setup`
- install nx cli globally `yarn global add @nrwl/cli`
- to start the next app run `yarn nx run client:serve`
- to start the nest app run `yarn nx run server:serve`
- to test husky pre-commit hook, you can break some eslint rules and try to commit.
  the commit will fail because of errors and warnings
- if you have prettier installed in your vscode, you should see codes formatted according to prettier rules configured in prettier.rc
- eslint rules should be enforced

### Any background context you want to provide?
 N/A

### What are the relevant tracker stories?
#3 

### Screenshots (if appropriate)
 N/A

### Questions:
 N/A
